packages/algob/src/lib/asc.ts

```diff
@@ -0,0 +1,33 @@
+import algosdk from "algosdk";
+
+import {
+  ASCDeploymentFlags
+} from "../types";
+
+async function getSuggestedParams (algocl: algosdk.Algodv2): Promise<algosdk.SuggestedParams> {
```

Comment:
Do not duplicate the functions. This already exists in `asa.ts`. Why would you duplicate the logic? Reuse. 
Move logic from `asa.ts` to a different file and use it from both places.

Reply:
I was getting fail transaction when i was using the tx logic, but using functions directly from algosdk works, but by calling tx.function() transaction was failing

Reply:
Then there is some kind of a difference between these two code chunks. What is it?

Reply:
asa.ts code calls function from tx.ts file, and asc.ts calls function from algosdk.
and tx file is in lib folder

Reply:
So why couldn't you reuse `getSuggestedParams` and added your own? Why do you need two implementations that are exactly the same?

Reply:
asa.ts calls function from tx.ts and using same for ASC was giving me genesis hash error for this transaction, so i checked the algorand docs, they were using algocl.getsuggestedparams() instead of tx.getsuggestedparams() and that worked.

Reply:
these are not exactly same

Reply:
the parameters that is passed one is tx.AlgoV2, and other one is algosdk.AlgoV2

Reply:
Really...?
`import tx from "algosdk";`
`import algosdk from "algosdk";`
You import the whole library and name it `algosdk` so they are the exact same thing

Reply:
It it would be this:
```
import { tx } from "algosdk";
import { algosdk } from "algosdk";
```
then they would be different. But they are not imported this way.

Reply:
` async function getSuggestedParams (algocl: tx.Algodv2): Promise<tx.SuggestedParams> {
  const params = await algocl.getTransactionParams().do();
  // Private chains may have an issue with firstRound
  if (params.firstRound === 0) {
    throw new Error("Suggested params returned 0 as firstRound. Ensure that your node progresses.");
    // params.firstRound = 1
  }
  return params;
} `

Yes, but
I was getting gensis hash error from this function when i debugged, 
I will check again ..

Reply:
got it, removed the double implementation

Reply:
Sorry, i'm not getting too much into this comments, but I agree with Martynas, that if possible, would be great to no duplicate the code. PS: You can always update other code ;)